tests/foreman/cli/test_discoveryrule.py

```diff
@@ -520,58 +386,37 @@ def test_negative_update_name(self, name, discoveryrule_factory):
         :parametrized: yes
         """
         rule = discoveryrule_factory()
-        with pytest.raises(CLIReturnCodeError):
-            DiscoveryRule.update({'id': rule.id, 'name': name})
-
-    @pytest.mark.tier3
-    def test_negative_update_hostname(self, discoveryrule_factory):
-        """Update discovery rule host name using number as a value
-
-        :id: c382dbc7-9509-4060-9038-1617f7fef038
-
-        :expectedresults: Rule host name is not updated
-
-        :CaseImportance: Medium
-
-        :CaseLevel: Component
-        """
-        rule = discoveryrule_factory()
-        with pytest.raises(CLIReturnCodeError):
-            DiscoveryRule.update({'id': rule.id, 'hostname': '$#@!*'})
-
-    @pytest.mark.tier3
-    def test_negative_update_limit(self, discoveryrule_factory):
-        """Update discovery rule host limit using invalid values
-
-        :id: e3257d8a-91b9-406f-bd74-0fd1fb05bb77
-
-        :expectedresults: Rule host limit is not updated
-
-        :CaseLevel: Component
-
-        :CaseImportance: Medium
-        """
-        rule = discoveryrule_factory()
+        priority = gen_string('alpha')
         host_limit = gen_string('alpha')
+        params = {
+            'name': name,
+            'hostname': '$#@!*',
+            'hosts-limit': host_limit,
+            'priority': priority,
+        }
+        key = random.choice(list(params.keys()))
         with pytest.raises(CLIReturnCodeError):
-            DiscoveryRule.update({'id': rule.id, 'hosts-limit': host_limit})
-
-    @pytest.mark.tier3
-    def test_negative_update_priority(self, discoveryrule_factory):
-        """Update discovery rule priority using invalid values
+            target_sat.cli.DiscoveryRule.update(
+                {
+                    'id': rule.id,
+                    key: params[key],
+                }
+            )
 
-        :id: 0778dd00-aa19-4062-bdf3-752e1b546ec2
+    @pytest.mark.tier1
+    def test_positive_delete(self, discoveryrule_factory, target_sat):
+        """Delete existing Discovery Rule
```

Comment:
I'd encourage to have a teardown to delete rules in all above tests instead of separate test for it, wdyt?

Reply:
yeah I can handle it with upcoming PR's